Robust.Shared/GameObjects/EntityManager.Components.cs

```
@@ -1391,6 +1391,141 @@ public CompRegistryEntityEnumerator CompRegistryQueryEnumerator(ComponentRegistr
+        }
+
+        /// <inheritdoc />
+        public HashSet<WeakEntityReference> GetWeakReferenceSet(HashSet<EntityUid> set)
+        {
+            var outSet = new HashSet<WeakEntityReference>(set.Count);
+            foreach (var element in set)
+            {
+                outSet.Add(GetWeakReference(element));
+            }
+            return outSet;
+        }
+
+        /// <inheritdoc />
+        public EntityUid? Resolve(WeakEntityReference weakRef)
+        {
+            if (weakRef.Entity != EntityUid.Invalid
+                && EntityExists(weakRef.Entity))
+            {
+                return weakRef.Entity;
+            }
+
+            return null;
+        }
+
+        /// <inheritdoc />
+        public EntityUid? Resolve(WeakEntityReference? weakRef)
+        {
+            return weakRef == null ? null : Resolve(weakRef.Value);
+        }
+
+        /// <inheritdoc />
+        public List<EntityUid> Resolve(List<WeakEntityReference> list)
+        {
+            var outList = new List<EntityUid>(list.Count);
+            foreach (var element in list)
+            {
+                if (TryGetEntity(element, out var ent))
+                    outList.Add(ent.Value);
+            }
+            return outList;
+        }
+
+        /// <inheritdoc />
+        public HashSet<EntityUid> Resolve(HashSet<WeakEntityReference> set)
+        {
+            var outSet = new HashSet<EntityUid>(set.Count);
+            foreach (var element in set)
+            {
+                if (TryGetEntity(element, out var ent))
+                    outSet.Add(ent.Value);
+            }
+            return outSet;
+        }
+
+        /// <inheritdoc />
+        public Entity<T>? Resolve<T>(WeakEntityReference<T> weakRef) where T : IComponent
+        {
+            if (weakRef.Entity != EntityUid.Invalid
+                && TryGetComponent(weakRef.Entity, out T? comp))
+            {
+                return new(weakRef.Entity, comp);
+            }
```

This should definitely do more than fail silently if the entity exists but does not have the component in question.